CSyncSDK/Operation.swift

```diff
@@ -196,8 +196,8 @@ class PubOperation : Operation
 			return nil
 		}
 
-		guard !key.isKeyPattern else {
-			error = err(CSError.invalidKey, msg:"Key may not contain wildcard characters")
+		guard !key.isKeyPattern || delete == true else {
+			error = err(CSError.invalidKey, msg:"Only delete pubs may contain a wildcard")
```

Comment:
I think this error message will be confusing to developers.  The developer does a "write", not a "pub", so a message about "delete pubs" won't help them.  I think you could leave the message as it is, but if you really feel it needs more detail, I would suggest "Key for write may not contain wildcard characters".